src/core/attributeformmodelbase.cpp

```diff
@@ -22,7 +22,15 @@
 #include <qgsrelationmanager.h>
 #include <qgsdatetimefieldformatter.h>
 #include <qgsvectorlayerutils.h>
-
+#include <qgis.h>
+#if VERSION_INT >= 31700
```

Comment:
@m-kuhn , the previous SDK used by the 1.8 branch is also 3.17dev, so that conditional check would lead to failure on patched QGIS 3.17 pre-attribute editor (which is what we're using).